### PR TITLE
feat: Added scheduling to CustomTrainingJob, CustomPythonPackageTrainingJob, CustomContainerTrainingJob

### PR DESCRIPTION
**Pull request by ivanmkc**
Added scheduling to CustomTrainingJob and CustomContainerTrainingJob according to established API from CustomJob.

- [x] CustomTrainingJob
- [x] CustomPythonPackageTrainingJob
- [x] CustomContainerTrainingJob
- [x] Unit tests
- [x] Integration test

### CODE REVIEW

File: tests/unit/aiplatform/test_training_jobs.py [hunk 1]
```diff
@@ -202,6 +206,17 @@ def _get_custom_job_proto_with_enable_web_access(state=None, name=None, version=
     return custom_job_proto
 
 
+def _get_custom_job_proto_with_scheduling(state=None, name=None, version="v1"):
+    custom_job_proto = copy.deepcopy(_TEST_BASE_CUSTOM_JOB_PROTO)
+    custom_job_proto.name = name
+    custom_job_proto.state = state
+
+    custom_job_proto.job_spec.enable_web_access = _TEST_ENABLE_WEB_ACCESS
```
Comment (morgandu) [inline tests/unit/aiplatform/test_training_jobs.py:214]: scheduling not enable_web_access?

File: tests/unit/aiplatform/test_training_jobs.py [hunk 2]
```diff
@@ -1397,6 +1502,47 @@ def test_run_call_pipeline_service_create_with_enable_web_access(
...
+    @pytest.mark.usefixtures(
+        "mock_pipeline_service_create_with_scheduling",
+        "mock_pipeline_service_get_with_scheduling",
+        "mock_get_backing_custom_job_with_scheduling",
+        "mock_python_package_to_gcs",
+    )
+    @pytest.mark.parametrize("sync", [True, False])
+    def test_run_call_pipeline_service_create_with_scheduling(self, sync, caplog):
+
+        caplog.set_level(logging.INFO)
```
Comment (morgandu) [inline tests/unit/aiplatform/test_training_jobs.py:1514]: no need caplog for testing, this applied to all 3 tests

File: tests/unit/aiplatform/test_training_jobs.py [hunk 3]
```diff
@@ -1397,6 +1502,47 @@ def test_run_call_pipeline_service_create_with_enable_web_access(
...
+            restart_job_on_worker_restart=_TEST_RESTART_JOB_ON_WORKER_RESTART,
+            sync=sync,
+        )
+
+        if not sync:
+            job.wait()
+
+        assert job._gca_resource == make_training_pipeline_with_scheduling(
+            gca_pipeline_state.PipelineState.PIPELINE_STATE_SUCCEEDED
+        )
```
Comment (morgandu) [inline tests/unit/aiplatform/test_training_jobs.py:1544]: should assert back_custom_job has the scheduling, this applied to all 3 tests

**Review by morgandu — CHANGES_REQUESTED** [review]

File: tests/unit/aiplatform/test_training_jobs.py (hunk 2 above)
Comment (ivanmkc) [reply tests/unit/aiplatform/test_training_jobs.py:1514]: I see this in multiple tests though.

Comment (ivanmkc) [reply tests/unit/aiplatform/test_training_jobs.py:1514]: Done

File: tests/unit/aiplatform/test_training_jobs.py (hunk 3 above)
Comment (ivanmkc) [reply tests/unit/aiplatform/test_training_jobs.py:1544]: Removed unneeded mock

File: tests/unit/aiplatform/test_training_jobs.py (hunk 1 above)
Comment (ivanmkc) [reply tests/unit/aiplatform/test_training_jobs.py:214]: Removed unneeded mock.

**Review by morgandu — APPROVED** [review]

**Comment by morgandu** [comment]
LGTM. Thanks!

**Pull request merged**